### DISCUSSION TITLE
[somename].loader.js not loading correctly from a docker container

### DISCUSSION THREAD

**Discussion by ljugtomten** [jeffreylanters/react-unity-webgl#200] [Help Wanted!] [answered]
Hello!

I have been fighting a strange issue for the past week and now I need some help. I recently upgraded to the latest version of this library.

When I serve my application from inside a docker container I can not get it to load. 

I am confrontend by the imfamous "Uncaught SyntaxError: Unexpected token '<'" in what appears to be the [somename].loader.js file 

![bild](https://user-images.githubusercontent.com/1965682/120665939-7e65d300-c48c-11eb-8d2a-ed0d8dc0eece.png)

The strange thing is that the unity.loader.js files shows up as the react index.html in some strange fashion so I get why it says "unexpected" token.

I have read tons of threads where people get similar errors because of compression settings, and I have tried all kinds of settings. Right now I am running uncompressed builds.

And the strange thing is that when I run the application outside a container it works fine.

I am using ASP NET 5 in the backend.

Thankful for any help or ideas!

br
/Henrik

**Comment by jeffreylanters** [comment, marked as answer]
Hi! You're serving your files incorrectly. Duplicate of #57

**Reply by jeffreylanters** [reply]
If you've any questions or need more help, feel free to reply on this thread. Best of luck with your project!

**Reply by jeffreylanters** [reply]
Almost forgot I've also added some more instructions on how to set up your UnityContext in the [documentation](https://github.com/jeffreylanters/react-unity-webgl#creating-a-unity-context-object). This might help you out as well.

> A most basic Unity Context instance consists of a Unity Config with four properties. The loaderUrl: a JavaScript file contain the Unity Loader that the web page needs in order to load the Unity content, the frameworkUrl: a JavaScript file containing the runtime and plugin code, the dataUrl: a custom file containg your Assets and scenes, and the codeUrl: a Web Assembly binary file containg native code. Sounds complicated? No worries, you don't need to remember any of this. Just add the paths to the files to the Unity Context, sit back and enjoy!
>
> The paths to these files are public paths and should be relative from your public root or index.html. They will be loaded during runtime due to their enourmous size, we do not want this in our bundle since this would have impact on our entire React Application. But no worries, you can use the built-in loading progression tracking in order to show some sort of loading screen.

**Reply by ljugtomten** [reply]
The unity files seem to be in the correct folder relative to the index.html file..... I am struggling to figure out what could be different in the container, when it is working ok when Im not running from a container...... The container is running linux.... could it be a windows vs linux difference somehow? (In running windows on my computer, but linux inside the container)

**Reply by ljugtomten** [reply]
I think I found it! 
The path to the unity files had an uppercase letter in one of the folders. It worked fine on windows, but not inside the linux containers! 😄

**Reply by jeffreylanters** [reply]
It might help you debug problems like this in the future, I noticed a HTML response in your screenshot while you were highlighting a JavaScript file, that indicates that your server probably returned a not found page, instead of your JavaScript. Glad you've found the problem! Best of luck with your project.

**Reply by jeffreylanters** [reply]
(Could you mark this thread as the valid answer, this way it might help other people?)

**Reply by ljugtomten** [reply]
Sure, thanks a bunch for the help! 😄

**Comment by Loaki07** [comment]
Recently, I was having the same issue with the app not working inside docker containers/ec2 servers.
However, the issue was a single character in the path (build file name) that was in uppercase. This does not seem to cause issues in windows/mac but in linux it does.

We need to be aware of the build file name changes when exported from unity and used in path for the react app.

I've created a demo to dockerize a react-unity-webgl build [here](https://github.com/Loaki07/docker-demo-react-unity-webgl).
